processors/base/writers/document_category.py

```diff
@@ -24,14 +24,29 @@ def write_document_category(conn, document_category):
     """
     create = False
     db = conn['database']
-    obj = db['document_categories'].find_one(id=document_category['id'])
-
-    # Create object
-    if not obj:
-        obj = {
-            'id': document_category['id'],
-        }
-        create = True
+
+    # This is used to sinchronize with `data_categories` of `data_contributions`
+    if document_category.get('id'):
+        obj = db['document_categories'].find_one(id=document_category['id'])
+        if not obj:
+            obj = {
+                'id': document_category['id'],
+            }
+            create = True
+
+    # This document category is not specific to data contributions
+    else:
+        obj = db['document_categories'].find_one(name=document_category['name'],
+            group=document_category['group'])
+        if not obj:
+            # Avoid ids that could be overwritten by syncronization with `data_categories`
+            max_id_query = 'SELECT max(id) FROM document_categories;'
```

Comment:
This is a hack to avoid letting `document_categories` from processors overwrite those that come from `data_categories`.  This is not a long term solution and it's only supposed to buy us some time until we merge API database with Explorer database.